src/promptflow/tests/sdk_cli_test/e2etests/test_experiment.py

```diff
@@ -0,0 +1,87 @@
+    FlowNode,
+)
+
+TEST_ROOT = Path(__file__).parent.parent.parent
+EXP_ROOT = TEST_ROOT / "test_configs/experiments"
+FLOW_ROOT = TEST_ROOT / "test_configs/flows"
+
+
+yaml = YAML(typ="safe")
+
+
+@pytest.mark.e2etest
+@pytest.mark.usefixtures("setup_experiment_table")
+class TestExperiment:
+    def test_experiment_from_template(self):
+        template_path = EXP_ROOT / "basic-no-script-template" / "basic.exp.yaml"
+        # Load template and create experiment
+        template = load_common(ExperimentTemplate, source=template_path)
+        experiment = Experiment.from_template(template)
+        # Assert experiment parts are resolved
+        assert len(experiment.nodes) == 2
+        assert all(isinstance(n, FlowNode) for n in experiment.nodes)
+        assert len(experiment.data) == 1
+        assert isinstance(experiment.data[0], ExperimentData)
+        assert len(experiment.inputs) == 1
+        assert isinstance(experiment.inputs[0], ExperimentInput)
+        # Assert type is resolved
+        assert experiment.inputs[0].default == 1
+        # Pop schema and resolve path
+        expected = dict(yaml.load(open(template_path, "r", encoding="utf-8").read()))
+        expected.pop("$schema")
+        expected["data"][0]["path"] = (FLOW_ROOT / "web_classification" / "data.jsonl").absolute().as_posix()
+        expected["nodes"][0]["path"] = (experiment._output_dir / "snapshots" / "main").absolute().as_posix()
+        expected["nodes"][1]["path"] = (experiment._output_dir / "snapshots" / "eval").absolute().as_posix()
+        experiment_dict = experiment._to_dict()
+        assert experiment_dict["data"][0].items() == expected["data"][0].items()
+        assert experiment_dict["nodes"][0].items() == expected["nodes"][0].items()
+        assert experiment_dict["nodes"][1].items() == expected["nodes"][1].items()
+        assert experiment_dict.items() >= expected.items()
+
+    def test_experiment_create_and_get(self):
+        template_path = EXP_ROOT / "basic-no-script-template" / "basic.exp.yaml"
+        # Load template and create experiment
+        template = load_common(ExperimentTemplate, source=template_path)
+        experiment = Experiment.from_template(template)
+        client = PFClient()
+        exp = client._experiments.create_or_update(experiment)
+        assert len(client._experiments.list()) > 0
+        exp_get = client._experiments.get(name=exp.name)
+        assert exp_get._to_dict() == exp._to_dict()
+
+    @pytest.mark.usefixtures("use_secrets_config_file", "recording_injection", "setup_local_connection")
+    def test_experiment_start(self):
+        template_path = EXP_ROOT / "basic-no-script-template" / "basic.exp.yaml"
+        # Load template and create experiment
+        template = load_common(ExperimentTemplate, source=template_path)
+        experiment = Experiment.from_template(template)
+        client = PFClient()
+        exp = client._experiments.create_or_update(experiment)
```

Comment:
Is there a way to start an expriment from experiment template directly? like `client._experiment.create_or_update(experiment_template="xxx")`